lib/init/npm-utils.js

```diff
@@ -119,14 +119,14 @@ function check(packages, opt) {
         throw error;
     }
 
-    if (opt.devDependencies && typeof fileJson.devDependencies === "object") {
-        deps = deps.concat(Object.keys(fileJson.devDependencies));
-    }
-    if (opt.dependencies && typeof fileJson.dependencies === "object") {
-        deps = deps.concat(Object.keys(fileJson.dependencies));
-    }
+    ["dependencies", "devDependencies"].forEach(key => {
+        if (opt[key] && typeof fileJson[key] === "object") {
+            deps = deps.concat(Object.keys(fileJson[key]));
```

Comment:
Since we're refactoring here anyway... It looks like `deps` is declared as an array, but the only uses are this concatenation line, and an `includes` check further below.

Why don't we just convert this to a Set? It seems that's what we're using it for anyway, so we might as well not have linear searches with each `includes` check further down.

And in that case, it will be more clear semantically that the order isn't important.

Reply:
:+1: